skbio/io/fasta.py

```diff
@@ -557,6 +563,10 @@
 
 @register_sniffer('fasta')
 def _fasta_sniffer(fh):
+
+    if _count_blank_lines(fh) > 5:
```

Comment:
There's also some duplication between FASTA and FASTQ readers related to handling of blank/whitespace-only lines at the beginning of the file that could be refactored into `_line_generator`, and I think we can push the duplicated sniffer logic into `_line_generator` as well.

How about removing `_count_blank_lines` and updating `_line_generator` to accept a keyword argument `skip_blanks=False`? If `False` (by default, the safest behavior), lines will be stripped and yielded (this is the current implementation). If `True`, only non-blank lines will be yielded. This should allow the duplicate logic to be removed from the FASTA and FASTQ readers.

For sniffing, we could add another keyword argument to indicate the maximum number of consecutive blank lines to skip if `skip_blanks=True`. The default behavior would place no restrictions on this number (necessary for the readers), and the sniffers could pass 5. If this number is exceeded, the function could either raise an error (which the sniffers could catch and handle), or it could just start yielding lines at this point and the sniffers would check the next line to determine whether to keep sniffing.

I'm not too concerned about having this number (5) duplicated across the two sniffers since we could conceivably want a different number to be passed for different formats (though 5 seems reasonable for both FASTA and FASTQ).

Does this factoring make sense?